src/PSRule.Rules.Azure/rules/Azure.Resource.Rule.ps1

```diff
@@ -34,3 +34,48 @@ Rule 'Azure.ResourceGroup.Name' -Ref 'AZR-000168' -Type 'Microsoft.Resources/res
     # Can't end with period.
     $Assert.Match($PSRule, 'TargetName', '^[-\w\._\(\)]*[-\w_\(\)]$');
 }
+
+
+# Synopsis: Ensure all properties named used for setting a username within a deployment are expressions (e.g. an ARM function not a string)
+Rule 'Azure.Resource.adminUsername' -Ref 'AZR-000284' -Type 'Microsoft.Resources/deployments' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {  
```

Comment:
I think is rule might be better named as `Azure.Deployment.AdminUsername` and update casing to be consistant with how other rules are cased. Also update the naming of the .md file.

```suggestion
Rule 'Azure.Deployment.AdminUsername' -Ref 'AZR-000284' -Type 'Microsoft.Resources/deployments' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {
```